cli/src/semdep/parsers/poetry.py

```diff
@@ -0,0 +1,67 @@
+from pathlib import Path
```

Comment:
Please add a toplevel comment explaining the highlevel goal of this module.
Also it would be great to add a link to the spec of this poetry file, so one can
know how you ended up with the grammar below (or a link to a blog post/whatever
trying to explain the poetry lock file format).

Reply:
I will definitely add more documentation. Unfortunately there is no spec for the poetry.lock format, nor have I found any third party explanation of the format in depth. This is the case for basically all the lockfile formats, they're woefully undocumented... I came to this grammar by looking at some examples basically. That's also why initial lockfile parsers tend to fail a lot. There's no documentation and I can only look at so many examples :(